Katello/katello
# Fixes #36267 - Audit repo changes and repo publication changes

#### What are the changes introduced in this pull request?
1. Audit repository addition/removal from CV.
2. Audit new publications on library repository.
3. Helper methods to query audits for CV
4. Helper method + rabl for needs_publish? boolean
#### Considerations taken when implementing this change?

#### What are the testing steps for this pull request?
1. Create a CV
2. Check needs_publish? => should be true
3. Add a few repos.
4. Publish.
5. Check needs_publish? => should be false immediately after publish
6. Remove/add some repos to CV
7. Check needs_publish? => should be true
8. Publish
9. Check needs_publish? => should be false immediately after publish
10. Remove some content from some repos
11. Check needs_publish? => should be true
12. Publish
13. Check needs_publish? => should be false immediately after publish
14. Force sync a repo. This creates a new publication.
15. Check needs_publish? => should be true

Test other flows around repositories and content in repositories that should or shouldn't require a publish on CV and check if needs_publish reflects that accurately.
